### PR TITLE
feat: [SRM-11980]: User should be able to delete manual query

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

Delete Manul Query Datadog and GCO metric

#### Screenshots
Screenshot 2022-09-20 at 1 16 32 PM

Screenshot 2022-09-20 at 1 18 49 PM

Screenshot 2022-09-20 at 1 19 00 PM
Screenshot 2022-09-20 at 1 16 40 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 14 Issues 
* Code issue 14 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11324)

File: src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx [hunk 1]
```diff
@@ -375,6 +390,32 @@ export default function MetricDashboardWidgetNav<T>(props: MetricDashboardWidget
     }
   }, [])
 
+  const deleteManualMetric = (id: string) => {
+    if (navContent[0]?.nodeData?.type === NodeType.MANUAL_INPUT_QUERY) {
+      const nodeList = navContent[0].childNodes?.map(node => node.id)
```
Comment (harshilbits) [inline src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx:395]: we might need a null check here `navContent[0].`

File: src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx [hunk 2]
```diff
@@ -375,6 +390,32 @@ export default function MetricDashboardWidgetNav<T>(props: MetricDashboardWidget
...
+      const activeNode = navContent[0].childNodes?.find(node => node.isSelected)
+      navContent[0].childNodes = navContent[0].childNodes?.filter(node => node.id !== id)
+      if (defaultTo(nodeList?.length, 0) > 1 && activeNode?.id === id && navContent[0].childNodes?.[0]) {
+        navContent[0].childNodes[0].isSelected = true
+        onSelectMetric(
+          navContent[0].childNodes[0].id as string,
```
Comment (harshilbits) [inline src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx:401]: same.

File: src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx]: done

File: src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/MetricDashboardWidgetNav/MetricDashboardWidgetNav.tsx:401]: done

**Review by harshilbits — APPROVED** [review]

**Pull request merged**